ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -620,8 +618,9 @@ public static void process_final_updates(BeaconState state) {
     }
 
     // Rotate current/previous epoch attestations
-    state.setPrevious_epoch_attestations(state.getCurrent_epoch_attestations());
-    state.setCurrent_epoch_attestations(
-        new SSZList<>(PendingAttestation.class, MAX_ATTESTATIONS * SLOTS_PER_EPOCH));
+    state.getPrevious_epoch_attestations().setAll(state.getCurrent_epoch_attestations());
```

Comment:
Similar to a copy by referencing the same immutable value, setting a field can also just reference the immutable subtree; no need to set all values.